packages/mattermost-redux/src/utils/post_list.ts

```diff
@@ -323,142 +322,76 @@ export function makeGenerateCombinedPost(): (state: GlobalState, combinedId: str
+            allUserIds.push(...userIds);
         }
     });
-
-    messageData.sort(comparePostTypes);
-
     function reduceUsers(acc: string[], curr: string) {
         if (!acc.includes(curr)) {
             acc.push(curr);
         }
         return acc;
     }
-
     return {
         allUserIds: allUserIds.reduce(reduceUsers, []),
         allUsernames: allUsernames.reduce(reduceUsers, []),
         messageData,
     };
 }
-
+function isUsersRelatedPost(postType: string) {
+    return (
+        postType === Posts.POST_TYPES.ADD_TO_TEAM ||
+        postType === Posts.POST_TYPES.ADD_TO_CHANNEL ||
+        postType === Posts.POST_TYPES.REMOVE_FROM_CHANNEL
+    );
+}
 export function combineUserActivitySystemPost(systemPosts: Post[] = []) {
     if (systemPosts.length === 0) {
         return null;
     }
-
-    const userActivities = systemPosts.reduce((acc: any, post: Post) => {
+    const userActivities: ActivityEntry[] = [];
+    systemPosts.reverse().forEach((post: Post) => {
         const postType = post.type;
-        let userActivityProps = acc;
-        const combinedPostType = userActivityProps[postType as string];
-
-        if (
-            postType === Posts.POST_TYPES.ADD_TO_TEAM ||
-            postType === Posts.POST_TYPES.ADD_TO_CHANNEL ||
-            postType === Posts.POST_TYPES.REMOVE_FROM_CHANNEL
-        ) {
-            const userId = post.props.addedUserId || post.props.removedUserId;
-            const username = post.props.addedUsername || post.props.removedUsername;
-            if (combinedPostType) {
-                if (Array.isArray(combinedPostType[post.user_id])) {
-                    throw new Error('Invalid Post activity data');
-                }
-                const users = combinedPostType[post.user_id] || {ids: [], usernames: []};
-                if (userId) {
-                    if (!users.ids.includes(userId)) {
-                        users.ids.push(userId);
-                    }
-                } else if (username && !users.usernames.includes(username)) {
-                    users.usernames.push(username);
-                }
-                combinedPostType[post.user_id] = users;
-            } else {
-                const users = {
-                    ids: [] as string[],
-                    usernames: [] as string[],
-                };
-
-                if (userId) {
-                    users.ids.push(userId);
-                } else if (username) {
-                    users.usernames.push(username);
-                }
-                userActivityProps[postType] = {
-                    [post.user_id]: users,
-                };
-            }
-        } else {
-            const propsUserId = post.user_id;
+        const actorId = post.user_id;
+        const userIds = isUsersRelatedPost(postType) ? post.props.addedUserId || post.props.removedUserId : [];
+        const usernames = isUsersRelatedPost(postType) ? post.props.addedUsername || post.props.removedUsername : [];
```

Comment:
To avoid mixing types and potentially confusing someone, could you change these to always be either strings or arrays and make the naming match that? As-is, their names are a bit misleading since they're plural which suggests that these would be arrays, but they can also be a string depending on the post type.

`userActivities` is correct as-is though since `CombinedSystemMessage` expects those to be arrays anyway

Reply:
Yes I agree , I will change them to be always arrays as the names states.